packages/x-data-grid/src/hooks/features/dimensions/useGridDimensions.ts

```diff
@@ -110,8 +111,9 @@ export function useGridDimensions(
   const groupHeaderHeight = Math.floor(
     (props.columnGroupHeaderHeight ?? props.columnHeaderHeight) * densityFactor,
   );
-  const headerFilterHeight = Math.floor(
-    (props.headerFilterHeight ?? props.columnHeaderHeight) * densityFactor,
+  const headerFilterHeight = Math.max(
+    Math.floor((props.headerFilterHeight ?? props.columnHeaderHeight) * densityFactor),
+    MIN_HEADER_FILTER_HEIGHT,
```

Comment:
What do we think about adding a minimum height for the header filter row? We need at minimum enough space for the inputs to not be clipped. Would fix #13048

Reply:
(there's a broken test for this, I'll fix if we want to move forward with this)

Reply:
Could we reduce the font sizes and spacing a bit to look it a bit more aligned with the compact density height (even if not exact at least closer)? I imagine it would be easier now that we have outlined text field.

![image](https://github.com/user-attachments/assets/efdfb3cf-fad8-447a-bb1f-6097021913fe)